web/src/hotspots.ts

```diff
@@ -57,27 +59,43 @@ const HOTSPOT_LOCATIONS = new Map([
     ],
 ]);
 
-const meta = {
+const meta: {
+    opened_hotspot_name: null | string;
+} = {
     opened_hotspot_name: null,
 };
 
-function compute_placement($elt, popover_height, popover_width, prefer_vertical_positioning) {
-    const client_rect = $elt.get(0).getBoundingClientRect();
+function compute_placement(
+    $elt: JQuery,
+    popover_height: number,
+    popover_width: number,
+    prefer_vertical_positioning: boolean,
+): Position {
+    // default to placing the popover in the center of the screen
+    let placement: Position = "viewport_center";
+
+    if (!$elt || $elt.length === 0) {
+        return placement;
+    }
```

Comment:
These should just be asserts; the app is broken if these don't exist, and placing the hotspot in the center in that case is probably wrong.